### DISCUSSION TITLE
Make Ctrl + u on Windows Work Like Linux

### DISCUSSION THREAD

**Discussion by wpalace** [wezterm/wezterm#6227] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows

### Which Wayland compositor or X11 Window manager(s) are you using?

n/a

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

I love that Wezterm gives me a consistent terminal experience across Windows and Linux. I have a habit of reaching for ctrl + u to clear everything from my cursor to the beginning of the current prompt but am frequently reminded that this does not work on Windows. How can I configure a key mapping to replicate this Linux behavior on Windows?

**Comment by loops** [comment]
Hi wpalace,

I don't use Windows, so can't give specific details for your environment, but Wezterm is not responsible for deciding to clear an input line or not.   All Wezterm does is pass the ctrl + u to the process that is running in the pane.  It's then up to the shell (or whatever is running) to clear the line or execute some other operation.    So this is really a question about how to configure your Windows environment to handle ctrl + u as you wish; not anything Wezterm can do.

The only way it could be a Wezterm problem, is if Wezterm is configured to use ctrl + u to trigger a local action, rather than as a regular key to pass on to the pane.    You can type:

     wezterm show-keys

To see if   CTRL  u     is being used that way.  (Only the first section "Default Key Table" matters).

**Comment by bew** [comment, marked as answer]
What @loops said, and in addition, if you are using PowerShell as your shell on Windows, you can put the following in your `$profile` file:
```powershell
Set-PSReadLineKeyHandler -Key Ctrl+u -Function BackwardDeleteLine
```
This will make Ctrl-u do the same as usual linux shells ;)

**Reply by wpalace** [reply]
Thank you, both. I do mostly use PowerShell when on Windows. Adding that cmdlet to my profile worked as described.
